Run tests on OSX

Run tests and see if there are any issues with running tests on osx

```
2022-01-05T00:05:59.792572Z 0 [ERROR] [MY-010267] [Server] The socket file path is too long (> 103): /private/var/folders/ks/w4v3szn557z1vkxyt8j52nzh0000gn/T/pytest-of-ben/pytest-18/pytest-mysql-mysql_proc_wg_static_db0/mysql.33307.sock
```

--- a/newsfragments/345.bugfix.rst
+++ b/newsfragments/345.bugfix.rst
@@ -1,0 +1,6 @@
+Raise exception with helpful message if unixsocket is too long on FreeBSD or MacOS system
+
+OSX gives out super long temp directories.  This isn't a problem until
+we run into an odd 103-character limit on the names of unix sockets
+`see this stackoverflow thread <https://unix.stackexchange.com/questions/367008/why-is-socket-path-length-limited-to-a-hundred-chars/367012#367012>`_.
+Here we warn and give the user a way out of it.

--- a/pytest_mysql/exceptions.py
+++ b/pytest_mysql/exceptions.py
@@ -17,5 +17,9 @@ class VersionNotDetected(PytestMySQLException):
         super().__init__("Could not detect version in {}".format(output))
 
 
+class SocketPathTooLong(PytestMySQLException):
+    """Exception raised the socket path is over 103 chars (mysql will fail to start)."""
+
+
 class DatabaseExists(PytestMySQLException):
     """Raise this exception, when the database already exists."""

--- a/pytest_mysql/exceptions.py
+++ b/pytest_mysql/exceptions.py
@@ -18,7 +18,11 @@ class VersionNotDetected(PytestMySQLException):
 
 
 class SocketPathTooLong(PytestMySQLException):
-    """Exception raised the socket path is over 103 chars (mysql will fail to start)."""
+    """
+    Exception raised the socket path is over 103 chars.
+
+    Raised on BSD/MacOS as Mysql will fail to start.
+    """
 
 
 class DatabaseExists(PytestMySQLException):

--- a/pytest_mysql/executor.py
+++ b/pytest_mysql/executor.py
@@ -1,15 +1,19 @@
 """Specified MySQL Executor."""
+import platform
 import re
 import subprocess
 from pathlib import Path
 from typing import Optional, Union, Literal, Any
 
-from mirakuru.base import SimpleExecutorType
 from pkg_resources import parse_version
 
 from mirakuru import TCPExecutor
 
-from pytest_mysql.exceptions import MySQLUnsupported, VersionNotDetected
+from pytest_mysql.exceptions import (
+    MySQLUnsupported,
+    VersionNotDetected,
+    SocketPathTooLong,
+)
 
 
 class MySQLExecutor(TCPExecutor):
@@ -42,7 +46,7 @@ class MySQLExecutor(TCPExecutor):
         :param params: string containing additional starting parameters
         :param base_directory: base directory where the temporary files,
             database files, socket and pid will be placed in.
-        :param user: mysql user name
+        :param user: mysql username
         :param host: server's host
         :param port: server's port
         :param timeout: executor's timeout for start and stop actions
@@ -138,6 +142,8 @@ class MySQLExecutor(TCPExecutor):
 
     def start(self) -> "MySQLExecutor":
         """Trigger initialisation during start."""
+        self._check_socket_path()
+
         implementation = self.implementation()
         if implementation == "mysql" and parse_version(
             self.version()
@@ -174,3 +180,9 @@ class MySQLExecutor(TCPExecutor):
         """Stop the server."""
         self.shutdown()
         return super().stop(*args, **kwargs)
+
+    def _check_socket_path(self) -> None:
+        if platform.system() in ["Darwin", "FreeBSD]"] and len(self.unixsocket) > 103:
+            raise SocketPathTooLong(
+                f"Socket path '{self.unixsocket}' is too long, please pass --basetemp=/tmp/pytest_mysql to pytest"
+            )

--- a/pytest_mysql/executor.py
+++ b/pytest_mysql/executor.py
@@ -182,7 +182,11 @@ class MySQLExecutor(TCPExecutor):
         return super().stop(*args, **kwargs)
 
     def _check_socket_path(self) -> None:
-        if platform.system() in ["Darwin", "FreeBSD]"] and len(self.unixsocket) > 103:
+        if (
+            platform.system() in ["Darwin", "FreeBSD]"]
+            and len(self.unixsocket) > 103
+        ):
             raise SocketPathTooLong(
-                f"Socket path '{self.unixsocket}' is too long, please pass --basetemp=/tmp/pytest_mysql to pytest"
+                f"Socket path '{self.unixsocket}' is too long, "
+                f"please pass ie. `--basetemp=/tmp/pytest_mysql` to pytest"
             )

